src/libraries/System.Runtime/tests/System/DelegateTests.cs

```diff
@@ -328,6 +330,32 @@ public static void DynamicInvoke_DefaultParameter_DateTimeParameterWithMissingVa
                 (DateTime)(new DateTimeWithDefaultValueAttribute(DateTimeMethod)).DynamicInvoke(new object[] { Type.Missing }));
         }
 
+        // CoreCLR prioritizes DateTimeConstant over CustomConstant attribute
+        [Fact]
+        public static void DynamicInvoke_DateTimeAndCustomConstantAttribute_DateTimeParameterWithMissingValue()
+        {
+            Assert.Equal(
+                new DateTime(42),
+                (DateTime)(new DateTimeDelegateWithDateTimeAndCustomConstantAttribute(DateTimeMethod)).DynamicInvoke(new object[] { Type.Missing }));
+        }
+
+        // CoreCLR prioritizes DateTimeConstant over CustomConstant attribute
+        [Fact]
+        public static void DynamicInvoke_CustomConstantAndDateTimeAttribute_DateTimeParameterWithMissingValue()
+        {
+            Assert.Equal(
+                new DateTime(42),
+                (DateTime)(new DateTimeDelegateWithCustomConstantAndDateTimeAttribute(DateTimeMethod)).DynamicInvoke(new object[] { Type.Missing }));
+        }
```

Comment:
@jkotas This newly added test started failing on NativeAOT lanes. Shall I try to align the implementation in NativeAOT code as well, or disable the test for NativeAOT and create a separate issue/PR which would reflect this?

Reply:
Since you have the full context on this, it would be nice to address the NativeAOT ordering as part of this PR if possible. Thank you!